autoload/ensime.vim.py

```diff
@@ -11,606 +11,10 @@ def EnsimeInitPath():
-
-    def com_en_declaration_split(self, args, range = None):
-        self.with_current_client(lambda c: c.open_declaration_split(args, range))
-
-    def com_en_symbol(self, args, range = None):
-        self.with_current_client(lambda c: c.symbol(args, range))
-
-    def com_en_inspect_type(self, args, range = None):
-        self.with_current_client(lambda c: c.inspect_type(args, range))
-
-    def com_en_doc_uri(self, args, range = None):
-        self.with_current_client(lambda c: c.doc_uri(args, range))
-
-    def com_en_doc_browse(self, args, range = None):
-        self.with_current_client(lambda c: c.doc_browse(args, range))
-
-    def com_en_suggest_import(self, args, range = None):
-        self.with_current_client(lambda c: c.suggest_import(args, range))
-
-    def com_en_set_break(self, args, range = None):
-        self.with_current_client(lambda c: c.set_break(args, range))
-
-    def com_en_clear_breaks(self, args, range = None):
-        self.with_current_client(lambda c: c.clear_breaks(args, range))
-
-    def com_en_debug_start(self, args, range = None):
-        self.with_current_client(lambda c: c.debug_start(args, range))
-
-    def com_en_classpath(self, args, range = None):
-        self.with_current_client(lambda c: c.en_classpath(args, range), True, False, True)
-
-    def com_en_debug_continue(self, args, range = None):
-        self.with_current_client(lambda c: c.debug_continue(args, range))
-
-    def com_en_backtrace(self, args, range = None):
-        self.with_current_client(lambda c: c.backtrace(args, range))
-
-    def com_en_clients(self, args, range = None):
-        for path in self.client_keys():
-            self.__message("{}: {}".format(path, self.client_status(path)))
-
-    def au_vimleave(self, filename):
-        self.teardown()
-
-    def au_buf_enter(self, filename):
-        self.with_current_client(lambda c: c.buffer_enter(filename), True, True)
-
-    def au_buf_leave(self, filename):
-        self.with_current_client(lambda c: c.buffer_leave(filename))
-
-    def au_buf_write_post(self, filename):
-        self.with_current_client(lambda c: c.type_check(filename))
-
-    def au_cursor_hold(self, filename):
-        self.with_current_client(lambda c: c.on_cursor_hold(filename))
-
-    def au_cursor_moved(self, filename):
-        self.with_current_client(lambda c: c.cursor_moved(filename), True, True)
 
-    def fun_en_complete_func(self, findstart, base = None):
-        if base == None:
-            base = findstart[1]
-            findstart = findstart[0]
-        if self.is_scala_file():
-            return self.with_current_client(lambda c: c.complete_func(findstart, base))
-        else:
-            return []
-    def on_receive(self, name, callback):
-        self.with_current_client(lambda c: c.on_receive(name,  callback))
-    def send_request(self, request):
-        self.with_current_client(lambda c: c.send_request(request))
+from ensime_shared.ensime import Ensime
```

Comment:
migration to a shared package probably should have been done in a separate PR so that the features you added are easier to see.